sdk/core/Azure.Core/src/Serialization/ModelSerializer.cs

```diff
@@ -162,7 +162,7 @@ private static IModelSerializable<object> GetInstance([DynamicallyAccessedMember
             return model;
         }
 
-        private static IModelSerializable<T> GetInstance<[DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicConstructors | DynamicallyAccessedMemberTypes.NonPublicConstructors)] T>() where T : IModelSerializable<T>
+        private static IModelSerializable<T> GetInstance<T>() where T : IModelSerializable<T>
```

Comment:
Why isn't this attribute necessary? It calls `GetObjectInstance(typeof(T))`, which needs the annotation.

Reply:
no warning is given.  I was using that as my guide.  I also originally thought it would be needed.

Reply:
Needed to use PublishTrimmed = true vs PublishAot = true in my csproj as well as set up a self contained app publish profile [following these instructions](https://learn.microsoft.com/en-us/dotnet/core/deploying/trimming/trim-self-contained#publish-with-visual-studio) in order to get the warning to show up.  Added these back.